charts/metering-ansible-operator/templates/crds/hive.crd.yaml

```diff
@@ -209,3 +209,41 @@ spec:
                       This should be a sub-directory of the "location" field.
                     minLength: 1
                     format: uri
+          oneOf:
+          - required:
+            - numBuckets
+            - clusteredBy
+          - required:
+            - external
+            - location
+            properties:
+              external:
+                enum:
+                - true
+          - required:
+            - managePartitions
+            - partitions
+            properties:
+              managePartitions:
+                enum:
+                - true
+              partitions:
+                type: array
+          # note: allow the minimal configuration, and enforce the schemas defined above
+          - properties:
```

Comment:
@chancez I wasn't able to think of anything that still enforced that the dependent properties were set together w/o listing the negation of those properties. While this `oneOf` case isn't very intuitive, it does enforce the sub-schemas listed above, while allowing the minimal configurations still.

Another thing to note is that [this would violate a structural schema](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#status-subresource)(as it includes `maxItems` outside of the root), but every CRD that's been updated recently also does this.

Reply:
Why use maxItems when we can just do `not: required: - clusteredBy`?